src/pyupgrade/checks.rs

```diff
@@ -196,3 +199,50 @@ pub fn unnecessary_future_import(
     }
     None
 }
+
+/// U011
+pub fn unnecessary_lru_cache_params(
+    decorator_list: &[Expr],
+    target_version: PythonVersion,
+    imports: Option<&BTreeSet<&str>>,
+) -> Option<Check> {
+    for expr in decorator_list.iter() {
+        if let ExprKind::Call {
+            func,
+            args,
+            keywords,
+        } = &expr.node
+        {
+            if args.is_empty()
+                && helpers::match_name_or_attr_from_module(func, "lru_cache", "functools", imports)
+            {
+                // Ex) `functools.lru_cache()`
+                if keywords.is_empty() {
+                    return Some(Check::new(
+                        CheckKind::UnnecessaryLRUCacheParams,
+                        Range::from_located(expr),
+                    ));
+                }
+                // Ex) `functools.lru_cache(maxsize=None)`
+                if target_version >= PythonVersion::Py39 && keywords.len() == 1 {
+                    let KeywordData { arg, value } = &keywords[0].node;
+                    if arg.as_ref().map(|arg| arg == "maxsize").unwrap_or_default()
```

Comment:
@chammika-become - I made this a little more conservative (used `map` then `upwrap_or_default` instead of force `unwrap`).

Reply:
Great! I've learned a lot from the improvements you made. Thanks!